k8s/src/main/scala/io/buoyant/k8s/resources.scala

```diff
@@ -174,19 +189,22 @@ private[k8s] class NsListResource[O <: KubeObject: TypeReference, W <: Watch[O]:
 }
 
 private[k8s] class NsObjectResource[O <: KubeObject: TypeReference, W <: Watch[O]: TypeReference](
-  val client: Client,
-  listPath: String,
+  parent: Resource,
   objectName: String,
+  maybeListName: Option[String] = None,
```

Comment:
I have a slight preference for just calling this `listName`; that's a bit more consistent with how we name option params elsewhere, but definitely not formalized in any way.